clientv3/client.go

```diff
@@ -148,17 +155,10 @@ func (c *Client) dialSetupOpts(endpoint string, dopts ...grpc.DialOption) (opts
 	}
 	opts = append(opts, dopts...)
 
-	// grpc issues TLS cert checks using the string passed into dial so
-	// that string must be the host. To recover the full scheme://host URL,
-	// have a map from hosts to the original endpoint.
-	host2ep := make(map[string]string)
-	for i := range c.cfg.Endpoints {
-		_, host, _ := c.dialTarget(c.cfg.Endpoints[i])
-		host2ep[host] = c.cfg.Endpoints[i]
-	}
+	c.balancer.updateHost2ep(c.cfg.Endpoints, c.creds)
```

Comment:
I feel like this function shouldn't be used outside the balancer. Can it be called in `newSimpleBalancer` instead?